src/components/VmActions/index.js

```diff
@@ -154,8 +153,7 @@ class VmActions extends React.Component {
           button='btn btn-default'
           className='pficon pficon-screen'
           vm={vm}
-          usbFilter={config.get('usbFilter')}
-          userId={config.getIn(['user', 'id'])} />
```

Comment:
Because now the `ConsoleButton` pulls it from redux store directly.

Reply:
This is not working like that, second parameter of `mapDispatchToProps` function is `ownProps`, that mean - not props that was got from store through `connect` function. https://github.com/reduxjs/react-redux/blob/master/docs/api.md

Reply:
Right.  `mapDispatchToProps` second argument is ownProps which is the props sent by the parent component plus the props created by the `mapStateToProps` function.  So as long as the prop is sent by the parent or redux `connect`ed, it'll get through to the dispatch functions.  This is done in other web-ui components as well (which is where I got the idea).

It makes sense to me to put props that can be pulled from redux state as close to the component that uses it as possible.  In this case, the only use for **config**, **usbFilter** and **userId** in all of `VmDetails/index.js` is to pass data to `ConsoleButton`.

https://github.com/sjd78/ovirt-web-ui/blob/f99b93af30597ac9deb8468a99d3561ca12ca5e2/src/components/VmActions/ConsoleButton.js#L133-L143

Reply:
No, not `plus the props created by the mapStateToProps function`. As I said it not work like this, `ownProps` means not props + props created by the mapStateToProps function, it means just props. Thus, please, move it back to `VmActions`.

Here is example and explanation: https://stackoverflow.com/questions/41198842/what-is-the-use-of-the-ownprops-arg-in-mapstatetoprops-and-mapdispatchtoprops

Reply:
Bohdan - [1] is a much better explanation of what you're describing.  Thank you for catching it as I wasn't aware `connect()` worked in that matter and why.

But, that doesn't mean the solution is to put those props in the containing component who doesn't need to know anything about how the `ConsoleButton` needs to map data to actions.  Since the main problem with how I tried to do things is repeated function rebinding on state change causing performance issues, it's easier to provide that action's data via function arguments.  I'm making that change now.  That way the `ownProps` isn't an issue, AND the parent component doesn't need to know so much about the `ConsoleButton`.

[1] - https://github.com/reduxjs/redux/blob/master/docs/faq/DesignDecisions.md#no-asynch-in-mapDispatchToProps

Reply:
Right, good point.